### ISSUE TITLE
[Tests] Rare failure of basic/global_vars_test due to wrong errno

### ISSUE THREAD

**Issue by prp** [lsds/sgx-lkl#555]
The following test "basic-global_vars_test-(nonrelease)-(run-sw)-(8-ethreads)" sometimes fails as follows:
```
[    0.135952] EXT4-fs (vda): mounted filesystem without journal. Opts: 
fopen failed errno(2)
TEST FAILED
test: 1 failed
[    0.144929] reboot: Restarting system
```

This is odd because the code for the test expects the first `fopen` to return an `errno` of 2, which is what happens: https://github.com/lsds/sgx-lkl/blob/c09fde496d58fe4227bac2a5f0575ae7e602c027/tests/basic/global_vars_test/global_vars_test.c#L89

One guess would be that something else is changing `errno`?

(cc: @davidchisnall @SeanTAllen)

**Comment by prp** [comment]
```
Execution timeout: 60
Will run tests for run-sw
/opt/sgx-lkl-debug/bin/sgx-lkl-disk create --size=5M --docker=./Dockerfile sgx-lkl-rootfs.img
Using detected host DNS server in /etc/resolv.conf: 168.63.129.16
Creating sgx-lkl-rootfs.img from Dockerfile ./Dockerfile...
Building Docker image...
Creating disk image file from Docker container...
Succesfully created sgx-lkl-rootfs.img.
SGXLKL_VERBOSE=1 SGXLKL_KERNEL_VERBOSE=1 /opt/sgx-lkl-debug/bin/sgx-lkl-run-oe --sw-debug sgx-lkl-rootfs.img global_vars_test -a -b -c hello
[   SGX-LKL  ] SGX-LKL (OE) 0.2.0-dev (58fa34b) LKL 5.3.0 [DEBUG build (-O0)] [SOFTWARE DEBUG]
[   SGX-LKL  ] nproc=8 ETHREADS=8 CMDLINE="mem=32M"
[   SGX-LKL  ] HW TLS support: conf->fsgsbase=0
[   SGX-LKL  ] Registering disk 0 (path='sgx-lkl-rootfs.img', mnt='/', [RW    ])
[   SGX-LKL  ] No tap device specified, networking will not be available.
[   SGX-LKL  ] get_signed_libsgxlkl_path... result=/opt/sgx-lkl-debug/bin/../lib/libsgxlkl.so.signed
[   SGX-LKL  ] oe_create_enclave... result=0 (OE_OK)
[   SGX-LKL  ] sgxlkl_enclave_init(ethread_id=0)
[[  SGX-LKL ]] sgxlkl_enclave_init(): enter
[[  SGX-LKL ]] _sgxlkl_enclave_show_attribute(): enclave base=0x7fee80000000 size=2.0 GiB
[[  SGX-LKL ]] _sgxlkl_enclave_show_attribute(): enclave heap base=0x7fee80b41000 size=1024.0 MiB end=0x7feec0b41000
[[  SGX-LKL ]] sgxlkl_enclave_init(): calling _dlstart_c()
[[  SGX-LKL ]] _register_enclave_signal_handlers(): Registering OE exception handler...
[[  SGX-LKL ]] __libc_init_enclave(): calling enclave_mman_init()
[[  SGX-LKL ]] __libc_init_enclave(): calling _lthread_sched_init()
[[  SGX-LKL ]] startmain(): enter
[[  SGX-LKL ]] lkl_start_init(): calling register_lkl_syscall_overrides()
[[  SGX-LKL ]] lkl_start_init(): calling initialize_enclave_event_channel()
[[  SGX-LKL ]] lkl_start_init(): calling lkl_virtio_console_add()
[[  SGX-LKL ]] lkl_start_init(): kernel command line: 'mem=32M console=hvc0'
[[  SGX-LKL ]] lkl_start_init(): lkl_start_kernel() called
[    0.000000] Linux version 5.3.0+ (AzDevOps@cisca24ba0001X2) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #1 Mon Jul 6 07:27:35 UTC 2020
[    0.000000] memblock address range: 0x7feebe8c0000 - 0x7feec08c0000
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 8080
[    0.000000] Kernel command line: mem=32M console=hvc0 virtio_mmio.device=256@0x1000000:1
[    0.000000] Dentry cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.000000] Inode-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory available: 32176k/32768k RAM
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 4096
[    0.000000] lkl: irqs initialized
[    0.000000] clocksource: lkl: mask: 0xffffffffffffffff max_cycles: 0x1cd42e4dffb, max_idle_ns: 881590591483 ns
[    0.000000] lkl: time and timers initialized (irq2)
[    0.000560] printk: console [hvc0] enabled
[    0.000560] pid_max: default: 4096 minimum: 301
[    0.000560] Mount-cache hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.000560] Mountpoint-cache hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.001123] devtmpfs: initialized
[    0.006466] random: get_random_bytes called from 0x00007fee8065f7da with crng_init=0
[    0.007026] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.007026] futex hash table entries: 16 (order: -4, 384 bytes, linear)
[    0.007026] xor: automatically using best checksumming function   8regs     
[    0.007026] NET: Registered protocol family 16
[    0.010365] clocksource: Switched to clocksource lkl
[    0.010365] NET: Registered protocol family 2
[    0.010925] tcp_listen_portaddr_hash hash table entries: 256 (order: 0, 4096 bytes, linear)
[    0.010925] TCP established hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.010925] TCP bind hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.010925] TCP: Hash tables configured (established 512 bind 512)
[    0.010925] UDP hash table entries: 128 (order: 0, 4096 bytes, linear)
[    0.010925] UDP-Lite hash table entries: 128 (order: 0, 4096 bytes, linear)
[    0.010925] NET: Registered protocol family 1
[    0.011491] virtio-mmio: Registering device virtio-mmio.0 at 0x1000000-0x10000ff, IRQ 1.
[    0.011491] workingset: timestamp_bits=62 max_order=14 bucket_order=0
[    0.013741] NET: Registered protocol family 38
[    0.013741] io scheduler mq-deadline registered
[    0.013741] io scheduler kyber registered
[    0.028842] loop: module loaded
[    0.028842] device-mapper: ioctl: 4.40.0-ioctl (2019-01-18) initialised: dm-devel@redhat.com
[    0.029404] wireguard: WireGuard 0.0.20191219 loaded. See www.wireguard.com for information.
[    0.029404] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[    0.029966] NET: Registered protocol family 10
[    0.029966] Segment Routing with IPv6
[    0.029966] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[    0.030528] This architecture does not have kernel memory protection.
[    0.030528] Run /init as init process
[[  SGX-LKL ]] lkl_start_init(): lkl_start_kernel() finished
[[  SGX-LKL ]] lkl_start_init(): creating LKL termination thread
[[  SGX-LKL ]] create_lkl_termination_thread(): enter
[[  SGX-LKL ]] lkl_start_init(): calling lkl_mount_virtial()
[[  SGX-LKL ]] lkl_termination_thread(): enter
[[  SGX-LKL ]] lkl_termination_thread(): Performed LKL syscall to get host task allocated (pid=36)
[[  SGX-LKL ]] lkl_start_init(): calling init_random()
[[  SGX-LKL ]] init_random(): Adding entropy to entropy pool
[    0.031652] random: crng init done
[[  SGX-LKL ]] init_enclave_clock(): Setting enclave realtime clock
[[  SGX-LKL ]] wg0 has public key CWeMn+ohafgqvvYN/7ZLMeE82dIMFXEBJamLTacobhQ=
[    0.032802] virtio_blk virtio1: [vda] 10240 512-byte logical blocks (5.24 MB/5.00 MiB)
[    0.173051] software IO TLB: mapped [mem 0x7fef63722000-0x7fef67722000] (64MB)
[[  SGX-LKL ]] lkl_mount_disk(): lkl_mount_disk(dev="/dev/vda", mnt="/mnt/vda", ro=0)
[    0.174177] EXT4-fs (vda): mounted filesystem without journal. Opts: 
[[  SGX-LKL ]] lkl_mount_disks(): Set working directory: /
[[  SGX-LKL ]] libc_start_main_stage2(): Calling app main: global_vars_test
fopen failed errno(2)
fake test failed: errno=0 l_err_no=0
TEST FAILED
test: 1 failed
[[  SGX-LKL ]] lkl_terminate(): terminating LKL (exit_status=0)
[[  SGX-LKL ]] lkl_termination_thread(): termination thread unblocked
[[  SGX-LKL ]] lkl_termination_thread(): calling lkl_sys_chdir(/)
[[  SGX-LKL ]] display_mount_table(): ========= /proc/mounts ===========
/dev/vda / ext4 rw,relatime 0 0
devtmpfs /dev devtmpfs rw,relatime 0 0
tmpfs /dev/shm tmpfs rw,relatime 0 0
tmpfs /tmp tmpfs rw,relatime,mode=777 0 0
tmpfs /mnt tmpfs rw,relatime,mode=777 0 0
none /sys sysfs rw,relatime 0 0
tmpfs /run tmpfs rw,relatime,mode=700 0 0
proc /proc proc rw,relatime 0 0
[[  SGX-LKL ]] display_mount_table(): ==================================
[[  SGX-LKL ]] lkl_termination_thread(): calling lkl_umount_timeout("/", MNT_DETACH, 2000)
[[  SGX-LKL ]] display_mount_table(): /proc/mounts cannot be accessed
[[  SGX-LKL ]] lkl_termination_thread(): calling lkl_virtio_netdev_remove()
[[  SGX-LKL ]] lkl_termination_thread(): calling lkl_sys_halt()
[    0.184288] reboot: Restarting system
[[  SGX-LKL ]] lkl_termination_thread(): lthread_detach2() done
[   SGX-LKL  ] ethread (7: 0) init (0: 0 exit=0) ethread (4: 0) ethread (2: 0) ethread (3: 0) ethread (5: 0) ethread (1: 0) ethread (6: 0) 
[   SGX-LKL  ] oe_terminate_enclave... done
[   SGX-LKL  ] SGX-LKL-OE exit: exit_status=0
Test run completed with EXIT CODE 0
```

**Comment by prp** [comment]
Fixed by #592
